File: frontend/src/types/quiz.ts
export type QuizOption = {
    id: number;
    option_text: string;
}

export type QuizQuestion = {
    id: number;
    question_text: string;
    difficulty: string;
    options: QuizOption[];
}

export type SubmitAnswerResponse = {
    is_correct: boolean;
    correct_option_id: number;
    explanation: string;
    xp_gained: number;
    total_xp: number;
    level: number;
    current_streak: number;
}

File: backend/app/schemas/quiz.py
from pydantic import BaseModel


class QuizOptionRead(BaseModel):
    id: int
    option_text: str

    class Config:
        from_attributes = True


class QuizQuestionRead(BaseModel):
    id: int
    question_text: str
    difficulty: str
    options: list[QuizOptionRead]

    class Config:
        from_attributes = True


class SubmitQuizAnswerRequest(BaseModel):
    question_id: int
    option_id: int


class SubmitQuizAnswerResponse(BaseModel):
    is_correct: bool
    correct_option_id: int
    explanation: str | None
    
    xp_gained: int | None = None
    total_xp: int | None = None
    level: int | None = None
    current_streak: int | None = None